### PR TITLE
CONTRIBUTING.md: Give more concrete advice on writing bugs and PRs.

### PR DESCRIPTION
**Pull request by octo**
Everything in the file is a suggestion; I'm hoping for a discussion around how we want to handle contributions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
-
+# Contribution guidelines
+
+Thanks for taking the time to contribute to the [collectd
+project](https://collectd.org/)! This document tries to give some guidance to
+make the process of contributing to *collectd* as pleasant and possible.
+
+## Bug reports
+
+Please report bugs as [GitHub
```
Comment (rubenk) [inline CONTRIBUTING.md]: Perhaps we should turn this section into an issue template (https://help.github.com/articles/creating-an-issue-template-for-your-repository/) and refer to that?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
+Issues](https://github.com/collectd/collectd/issues). Try to answer the
+following questions:
+
+*   Which version of *collectd* are you using?
```
Comment (rubenk) [inline CONTRIBUTING.md]: Platform / distro would be helpful too.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
+
+Please monitor your issue for a couple of days and reply to questions. To keep
+the project manageable have to do some housekeeping, meaning we will close
+issues that have become stale.
+
+## Code contributions
+
+Please open a [GitHub Pull Request](https://github.com/collectd/collectd/pulls)
+(PR) to contribute bug fixes, features, cleanups, new plugins, … Patches sent to
+the mailing list have a tendency to fall through the cracks.
```
Comment (rubenk) [inline CONTRIBUTING.md]: I think we can turn this into a Pull Request template and refer to that (https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/)

**Comment by rubenk** [comment]
Nice, this is sorely needed IMHO.

README also contains some information at the bottom. When this document is done we should remove that and refer to this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (octo) [inline CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (octo) [inline CONTRIBUTING.md]: Good idea, added.

File: CONTRIBUTING.md (hunk 3 above)
Comment (octo) [inline CONTRIBUTING.md]: I think PRs differ greatly from issues: here, we want to communicate something *to the person* opening the PR, namely what we want the changes to include. I'm not very concerned about the exact wording of the PR message itself.

In issues, we want the user to communicate something *to us*, hence a checklist / template makes much more sense.

I've seen projects to checklists like

- [ ] manpage updated

but I'm not sure if that's all that helpful: not all changes require a manpage update, so a "no" doesn't tell you much. I can't really come up with a good checklist that would actually help me with a PR review.

Comment (rubenk) [inline CONTRIBUTING.md]: No you're right, I can't come up with a relevant checklist for a PR either. And most of the time, the code changes + commit messages speak for themselves.

Comment (rubenk) [inline CONTRIBUTING.md]: > Patches sent to the mailing list have a tendency to fall through the cracks.

Perhaps skip that sentence. If we don't mention it, hopefully people won't even try.

README still says you can send patches to the mailinglist but we should remove that.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,38 +1,56 @@
...
+    For existing files, please blend into surrounding code, i.e. mimic the
+    coding style of the code around your changes.
+*   *Documentation:* New config options need to be documented in two places: the
+    manpage (`src/collectd.conf.pod`) and the example config
+    (`src/collectd.conf.in`).
+*   *Continuous integration:* Once your PR is created, our continuous
+    integration environment will try to build it on a number of platforms. If
+    this reports a failure, please investigate and fix the problem. We will at
+    best do a very casual review for failing PRs.
+*   *Don't rebase:* Rebasing our branch destroys the review history. If a review
```
Comment (rubenk) [inline CONTRIBUTING.md]: s/our/your/

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,38 +1,56 @@
...
+    takes a long time, we may ask you to rebase on a more recent *master*, but
+    please don't do it without being asked.
+*   *types.db:* One of the most common mistakes done by new contributors is the
```
Comment (rubenk) [inline CONTRIBUTING.md]: s/done/made/

File: .github/ISSUE_TEMPLATE [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+(Description of the behavior / output that you observed)
+
+## Steps to reproduce
+
+*   step 1
+*   step 2
+*   step 3
+
+<!-- vim: set syntax=markdown : -->
```
Comment (rubenk) [inline .github/ISSUE_TEMPLATE]: Nice, and simple enough so people might actually fill it in ;)

One suggestion, could you name this ISSUE_TEMPLATE.md?

File: CONTRIBUTING.md (hunk 4 above)
Comment (octo) [inline CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 5 above)
Comment (octo) [inline CONTRIBUTING.md]: Done.

File: .github/ISSUE_TEMPLATE (hunk 6 above)
Comment (octo) [inline .github/ISSUE_TEMPLATE]: I copied the file name from the doc verbatim. Upon close reading, it turns out that they don't care about case and ignore filename extensions, so I've moved the file to `issue_template.md`.

File: CONTRIBUTING.md (hunk 3 above)
Comment (octo) [inline CONTRIBUTING.md]: Good point, done.

**Pull request merged**